src/strategy_manager/strategy.manager.cpp

```diff
@@ -0,0 +1,156 @@
+/******************************************************************************
+ * Copyright © 2013-2019 The Komodo Platform Developers.                      *
+ *                                                                            *
+ * See the AUTHORS, DEVELOPER-AGREEMENT and LICENSE files at                  *
+ * the top-level directory of this distribution for the individual copyright  *
+ * holder information and the developer policies on copyright and licensing.  *
+ *                                                                            *
+ * Unless otherwise agreed in a custom licensing agreement, no part of the    *
+ * Komodo Platform software, including this file may be copied, modified,     *
+ * propagated or distributed except according to the terms contained in the   *
+ * LICENSE file                                                               *
+ *                                                                            *
+ * Removal or modification of this copyright notice is prohibited.            *
+ *                                                                            *
+ ******************************************************************************/
+
+#include <vector>
+#include <unordered_map>
+
+#include "strategy.manager.hpp"
+
+namespace antara {
+  bool asset::operator==(const asset &other) const {
+    return symbol == other.symbol;
+  };
+
+  std::pair<antara::asset, antara::asset> pair::to_std_pair() {
+    return std::pair<antara::asset, antara::asset>(base, quote);
+  };
+
+  bool pair::operator==(const pair &rhs) const {
+    return base == rhs.base && quote == rhs.quote;
+  };
+}
+
+namespace std {
+
+  template <>
+  struct hash<antara::pair> {
+    std::size_t operator()(const antara::pair& p) const {
+      using std::size_t;
+      using std::hash;
+      using std::string;
+
+      std::size_t h1 = std::hash<std::string>{}(p.base.symbol);
+      std::size_t h2 = std::hash<std::string>{}(p.quote.symbol);
+
+      return h1 ^ (h2 << 1);
+    }
+  };
+}
+
+enum class Side { BUY, SELL, BOTH };
+
+class MMStrategy {
+public:
+  antara::pair pair;
+  float spread;
+  float quantity;
+  Side side;
+};
+
+class OrderLevel {
```

Comment:
If all the fields need to be public use a struct and remove public:

Reply:
I think order level should be a POD:

`A Plain Old Data Structure in C++ is an aggregate class that contains only PODS as members, has no user-defined destructor, no user-defined copy assignment operator, and no nonstatic members of pointer-to-member type.`

and used as order_level{price, quantity, side}

we should rename `OrderLevel` to `order_level`.

It's should be a struct.

Ignore this message if you plan to add member function but for the moment it's just seem's to be data.

Reply:
```cpp
struct order_level {
antara::st_price price;
antara::st_quantity quantity;
side side;
}

order_level ol{price, quantity, side}
```

Reply:
Yeah, chances are that its just a value object. I just made it an object until I know better.